pkg/services/alerting/notifiers/slack.go

```diff
@@ -30,7 +30,7 @@ func init() {
       <div class="gf-form max-width-30">
         <span class="gf-form-label width-8">Url</span>
         <div class="gf-form gf-form--grow" ng-if="!ctrl.model.secureFields.url">
-          <input type="text" required class="gf-form-input max-width-30" ng-model="ctrl.model.secureSettings.url" placeholder="Slack incoming webhook url"></input>
+          <input type="text" required class="gf-form-input max-width-30" ng-init="ctrl.model.secureSettings.url = ctrl.model.settings.url || ''; ctrl.model.settings.url = null;" ng-model="ctrl.model.secureSettings.url" placeholder="Slack incoming webhook url"></input>
```

Comment:
This feels like a dirty solution, but didn't have any other idea at the time. Idea being that editing an existing slack notification channel migrates url and token from settings to secure settings. It felt like a good idea, but this could be a potential problem for anyone upgrading to Grafana version with this feature and then reverts to an older Grafana version.

Reply:
Thanks for fixing editor :-)

As you pointed out, potential problem exists if user update setting with this feature and then downgrade Grafana.
But if we care about the case, we need to double write url/token to settings and secureSettings.

I think user notice about the incompatibility, user will re-set url and token after downgrading.
I suggest two option.

1. Simply notice about possible problem in slack setting UI.
2. Double write url to settings. I think securing URL is low priority, but url is important to work slack notification. Stop double write url after few months. (don't double write token)

Reply:
```suggestion
          <input type="text" required class="gf-form-input max-width-30" ng-init="ctrl.model.secureSettings.url = ctrl.model.settings.url || ''" ng-model="ctrl.model.secureSettings.url" placeholder="Slack incoming webhook url"></input>
```